src/gdal/mod.rs

```diff
@@ -0,0 +1,67 @@
+//! `erased-cells` connector to GDAL I/O
+
+use crate::error::Result;
+use crate::{with_ct, CellEncoding, CellType, NoData};
+use gdal::raster::GdalDataType;
+use num_traits::ToPrimitive;
+use paste::paste;
+
+mod rasterband;
+pub use rasterband::*;
+
+/// Convert from [`GdalDataType`] to appropriate [`CellType`].
+impl TryFrom<GdalDataType> for CellType {
+    type Error = crate::error::Error;
+
+    fn try_from(value: GdalDataType) -> Result<Self, Self::Error> {
+        match value {
+            GdalDataType::UInt8 => Ok(CellType::UInt8),
+            GdalDataType::UInt16 => Ok(CellType::UInt16),
+            GdalDataType::Int16 => Ok(CellType::Int16),
+            GdalDataType::UInt32 => Ok(CellType::UInt32),
+            GdalDataType::Int32 => Ok(CellType::Int32),
+            GdalDataType::Float32 => Ok(CellType::Float32),
+            GdalDataType::Float64 => Ok(CellType::Float64),
+            o => Err(Self::Error::UnsupportedCellTypeError(o.to_string())),
+        }
+    }
+}
+
+/// No-data conversion support.
+pub(crate) struct GdalND(Option<f64>);
```

Comment:
This helps convert GDAL's f64 encoding of all no-data values into our version that's a bit more type-safe.